packages/eas-json/src/utils.ts

```diff
@@ -73,4 +93,36 @@ export class EasJsonUtils {
     const easJson = await accessor.readAsync();
     return resolveSubmitProfile({ easJson, platform, profileName });
   }
+
+  public static getCacheSettings<T extends Platform>(
+    buildProfile: BuildProfile<T>,
+    ctx: any
+  ): Cache {
+    if (
+      buildProfile?.cache?.paths &&
+      buildProfile.cache.paths.length > 0 &&
+      buildProfile?.cache?.customPaths &&
+      buildProfile.cache.customPaths.length > 0
+    ) {
+      throw new InvalidEasJsonError(
+        `Found invalid cache settings in ${chalk.bold('eas.json')}. Cannot use both ${chalk.bold(
+          'paths'
+        )} and ${chalk.bold('customPaths')} at the same time`
+      );
+    }
+
+    const cacheSettings = {
+      ...EasJsonUtils.cacheDefaults,
+      ...buildProfile.cache,
+      clear: ctx.clearCache,
+    };
+
+    if (cacheSettings.customPaths) {
+      if (cacheSettings.customPaths.length > 0) {
+        cacheSettings.paths = cacheSettings.customPaths;
+      }
+      delete cacheSettings.customPaths;
+    }
+    return cacheSettings;
```

Comment:
This saves one level of indentation.
```suggestion
    if (!cacheSettings.customPaths) {
      return cacheSettings;
    }
    if (cacheSettings.customPaths.length > 0) {
      cacheSettings.paths = cacheSettings.customPaths;
    }
    delete cacheSettings.customPaths;
    return cacheSettings;
```